includes/multi-currency/class-currency-switcher-widget.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * WooCommerce Payments Currency Switcher Widget
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Multi_Currency;
+
+use WP_Widget;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Currency Switcher Widget Class
+ */
+class Currency_Switcher_Widget extends WP_Widget {
+
+	const DEFAULT_SETTINGS = [
+		'title'  => '',
+		'symbol' => true,
+		'flag'   => false,
+	];
+
+	/**
+	 * Multi-Currency instance.
+	 *
+	 * @var Multi_Currency
+	 */
+	protected $multi_currency;
+
+	/**
+	 * Register widget with WordPress.
+	 *
+	 * @param Multi_Currency $multi_currency The Multi_Currency instance.
+	 */
+	public function __construct( Multi_Currency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+
+		parent::__construct(
+			'currency_switcher_widget',
+			__( 'Currency Switcher', 'woocommerce-payments' ),
+			[ 'description' => __( 'Let your customers switch between your enabled currencies', 'woocommerce-payments' ) ]
+		);
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @param array $args     Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	public function widget( $args, $instance ) {
+		$instance = wp_parse_args(
+			$instance,
+			self::DEFAULT_SETTINGS
+		);
+
+		$title = apply_filters( 'widget_title', $instance['title'] );
```

Comment:
I don't know much about WP_Widgets, but it looks like most usages of this filter listed [here](https://developer.wordpress.org/reference/hooks/widget_title/) also include `$instance` and `$this->id_base`, like: `$title = apply_filters( 'widget_title', $instance['title'], $instance, $this->id_base );`.

Would you mind including them too? I think the `id_base` property could be used to filter which widget title is being modified by functions that plug into `widget_title`.

Reply:
Addressed in 24643f6